### ISSUE TITLE
Non ascii literal false positive

### ISSUE THREAD

**Issue by Milo123459** [rust-lang/rust-clippy#7307]
I tried this code:

So long so I put it in a bin: https://sourceb.in/KycdLRQdfR

I expected to see this happen: *explanation* Clippy shouldn't throw an error about non ascii literals, even though there are none.

Instead, this happened: *explanation*

```
error: literal non-ASCII character detected
    --> src\utils\mod.rs:808:11
     |
808  |       async fn test_content_safe() {
     |  ___________^
809  | |         use std::{collections::HashMap, sync::Arc};
810  | |
811  | |         use chrono::{DateTime, Utc};
...    |
1035 | |         assert_eq!(with_here_mention, content_safe(&cache, with_here_mention, &options).await);
1036 | |     }
     | |_____^
     |
note: the lint level is defined here
    --> src\lib.rs:59:5
     |
59   |     clippy::non_ascii_literal,
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#non_ascii_literal 
     = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
error: aborting due to 47 previous errors; 2 warnings emitted

error: could not compile `serenity`
```

### Meta

- `cargo clippy -V`: e.g. clippy 0.1.52 (9bc8c42b 2021-05-09)
- `rustc -Vv`:
  ```
  rustc 1.52.1 (9bc8c42bb 2021-05-09)
  binary: rustc
  commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
  commit-date: 2021-05-09
  host: x86_64-pc-windows-msvc
  release: 1.52.1
  LLVM version: 12.0.0
  ```

<details><summary>Backtrace</summary>
  <p>
  
  ```
  <backtrace>
  ```
  
  </p>
</details>

**Comment by giraffate** [comment]
Can you have the minimal reproduction on the playground?

**Comment by TonalidadeHidrica** [comment]
Maybe this is a duplicate of #7739.
